src/conversions.jl

```diff
@@ -705,6 +694,32 @@ function convert_attribute(ls::Symbol, ::key"linestyle")
     end
 end
 
+function pattern(linestyle, modifier=:normal; kwargs...)
+    float.([0; cumsum(_pattern(linestyle, modifier; kwargs...))])
+end
+
+"The linestyle patterns are inspired by the LaTeX package tikZ as seen here https://tex.stackexchange.com/questions/45275/tikz-get-values-for-predefined-dash-patterns."
+function _pattern(linestyle, modifier=:normal;
+                  dot = 1,  dot_gap  = (normal = 2, dense = 1, loose = 4),
+                  dash = 3, dash_gap = (normal = 3, dense = 2, loose = 6)
+                  )
+
+    if linestyle == :dash
+        gap = getproperty(dash_gap, modifier)
```

Comment:
took me a bit to understand the logic here, could we simplify that somehow? also, how can anybody actually set the modifier? you just call `pattern(ls)`

Reply:
Sorry, we could definitely hard-code the few patters that we have ;-)

But we could also use this function to allow more flexible patterns, that is any combination of dash and dot. And we could also make use of the modifiers. E.g. `("-....-.", :loose)`

Reply:
not necessarily hardcode, but I didn't see a path for a user to set the modifier. Also I like the idea with "-..-" etc, but you didn't add that yet as far as I can see?

Reply:
You're right, it's not yet there.

Reply:
ok I was confused after you mentioned that on Slack :)